cmd/evm/internal/t8ntool/transition.go

```diff
@@ -252,11 +252,20 @@ func Transition(ctx *cli.Context) error {
 			return NewError(ErrorConfig, errors.New("EIP-1559 config but missing 'currentBaseFee' in env section"))
 		}
 	}
-	// Sanity check, to not `panic` in state_transition
-	if prestate.Env.Random != nil && !chainConfig.IsLondon(big.NewInt(int64(prestate.Env.Number))) {
-		return NewError(ErrorConfig, errors.New("can only apply RANDOM on top of London chainrules"))
-	}
-	if env := prestate.Env; env.Difficulty == nil {
+
+	if env := prestate.Env; chainConfig.IsMerged(big.NewInt(int64(env.Number))) {
```

Comment:
This `IsMerged` makes the semantics simple, but I could skip that method and just do 
```
isMerged = (env.Random == nil) && (env.Difficulty = nil || env.Difficulty.BitLen()==0)
```
Would that be preferred?

Reply:
Yes, I think I prefer this over making decisions based on a fork that "we should not base any changes other than the forkid" on.